service/src/main/java/uk/gov/hmcts/reform/fpl/utils/QueryManagementUtils.java

```diff
@@ -0,0 +1,224 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.enums.CaseRole;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.UserService;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+
+@Slf4j
+public class QueryManagementUtils {
+
+    private QueryManagementUtils() {
+
+    }
+
+    public static String getCurrentCollectionByUserService(CaseData caseData,
+                                        UserService userService,
+                                        Map<CaseRole, String> collectionMapping) {
```

Comment:
can we refactor this method into a service, let say QueryManagmentService? So instead of injecting userService and passing it as param, those controller can simply inject the QueryManagmentService bean